Dockerfile

```diff
@@ -54,18 +56,20 @@ RUN set -ex \
  && ln -sf /dev/stderr /var/log/nginx/error.log \
  && curl -s https://getcomposer.org/installer | php \
  && mv composer.phar /usr/local/bin/composer \
- && composer selfupdate 2.2.18
+ && composer selfupdate 2.2.18 \
```

Comment:
composer could be installed through an intermediate Docker container, for example:

```
COPY --from=composer:2.2.18 /usr/bin/composer /usr/local/bin/composer
```

Reply:
I tried and the image size is the same. The Docker image build process is a bit slower with your solution because the Composer Docker image has many layers. You can open another PR if you want.

Reply:
Noted